lib/models/custom-apps/custom-app.models.ts

```diff
@@ -0,0 +1,62 @@
+import { CustomAppsContracts, SharedContracts } from '../../contracts';
+import { SharedModels } from '../shared/shared-models';
+
+export namespace CustomAppModels {
+    export class CustomApp implements SharedModels.IBaseModel<CustomAppsContracts.ICustomAppContract> {
+        public name: string;
+        public codename: string;
+        public source_url: string;
+        public config: string | null;
+        public allowed_roles?: SharedContracts.ICodenameIdReferenceContract[];
+        public _raw!: CustomAppsContracts.ICustomAppContract;
+
+        constructor(data: {
+            name: string;
+            codename: string;
+            source_url: string;
+            config: string | null;
+            allowed_roles?: SharedContracts.ICodenameIdReferenceContract[];
+            _raw: CustomAppsContracts.ICustomAppContract;
+        }) {
+            this.name = data.name;
+            this.codename = data.codename;
+            this._raw = data._raw;
+            this.source_url = data.source_url;
+            this.config = data.config;
+        }
+    }
+
+    export interface IAddCustomAppData {
+        name: string;
+        codename: string;
+        source_url: string;
+        config: string | null;
```

Comment:
config might probably be undefined

Reply:
Yeah, you're right here. When posting data, the `config` prop can be missing and the request still succeeds. Making it `undefined`.